### PR TITLE
doc: getting started: update SES configuration options

### PR DESCRIPTION
**Pull request by ru-fu**
Update documentation of SEGGER Embedded Studio configuration
options and screenshots.

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

### CODE REVIEW

**Comment by ru-fu** [comment]
We're missing information about the Tools > Options configuration.
Which fields are required there? Which are prefilled?
I can't get it to work here ... @tejlmand ?

File: doc/nrf/gs_programming.rst [hunk 1]
```diff
@@ -29,28 +29,29 @@ You can download it from the following links:
...
 	- **Build Directory** - the folder in which to run the build
-	- **Board Directory** - the location of the board description of the board
-	  for which to build the project
-	- **GNU ARM Embedded Toolchain Directory** - the location of the GNU Arm
-	  Embedded Toolchain installation
+	- **Delete Existing CMakeCache.txt** - select this option to ensure that you are not building with an outdated build cache
 
-   The following figure shows an example configuration for the Asset Tracker
-   sample built for the ``nrf9160_pca10090`` board target:
+   The following figure shows an example configuration for the Asset Tracker application built for the ``nrf9160_pca10090`` board target:
```
Comment (ru-fu) [inline doc/nrf/gs_programming.rst]: Does it make more sense to have nrf9160_pca10090ns as example?

File: doc/nrf/gs_programming.rst [hunk 2]
```diff
@@ -98,8 +99,12 @@ it might return an error indicating a project load failure. For example::
...
-project file paths.
+This issue might be caused by a variety of problems, such as incorrectly specified project file paths.
 SES helps you to identify the source of the issue by providing a text output
 with detailed information about the error. Make sure to click **OK** on the
 error pop-up message and then inspect the text output in SES.
+
+.. tip::
+   SES uses the PATH variable to find executables.
+   If you get an error that a tool or command cannot be found, add its location to the PATH variable.
+   For some tools, you can explicitly specify the location under **Tools -> Options** (select the **nRF Connect** tab).
```
Comment (ru-fu) [inline doc/nrf/gs_programming.rst]: Do we need to add the WEST path here for macOS?

File: doc/nrf/gs_programming.rst [hunk 3]
```diff
@@ -98,8 +99,12 @@ it might return an error indicating a project load failure. For example::
...
-This issue might be caused by a variety of problems, such as incorrectly specified
-project file paths.
+This issue might be caused by a variety of problems, such as incorrectly specified project file paths.
 SES helps you to identify the source of the issue by providing a text output
 with detailed information about the error. Make sure to click **OK** on the
 error pop-up message and then inspect the text output in SES.
+
+.. tip::
+   SES uses the PATH variable to find executables.
+   If you get an error that a tool or command cannot be found, add its location to the PATH variable.
```
Comment (ru-fu) [inline doc/nrf/gs_programming.rst]: @hakonfam @sigvartmh See https://devzone.nordicsemi.com/f/nordic-q-a/46173/can-t-flash-merge-hex-error - do we need to add a note that mergehex must be in the path?

Comment (sigvartmh) [reply doc/nrf/gs_programming.rst]: I think that could be a good idea after reading the support case.

File: doc/nrf/gs_programming.rst [hunk 4]
```diff
@@ -29,28 +29,29 @@ You can download it from the following links:
...
-	- **Zephyr Base** - the location of your cloned Zephyr repository
-	- **CMakeLists.txt** - the location of the :file:`CMakeLists.txt` project file
-	  of the sample that you want to work with
+   If you want to change these settings later, click **Tools -> Options** and select the **nRF Connect** tab.
+#. To import a project into SES, you must specify the following information:
+
+	- **CMakeLists.txt** - the location of the :file:`CMakeLists.txt` project file of the sample that you want to work with
+	- **Board Directory** - the location of the board description of the board for which to build the project
+	- **Board Name** - the board name (select from the list that is populated based on the board directory)
 	- **Build Directory** - the folder in which to run the build
```
Comment (tejlmand) [inline doc/nrf/gs_programming.rst]: should we put a note, that this field we be auto filled based upon the board name, but can be changed by user ?
As example, if user select `nrf9160_pca10090ns`, the build folder will be pre-filled with:
`<path-to-app>/build_nrf9160_pca10090ns`

File: doc/nrf/gs_programming.rst (hunk 2 above)
Comment (tejlmand) [reply doc/nrf/gs_programming.rst]: It is currently being discussed on how to improve the user experience on MacOS.
But for time being I believe i would be a good idea to descibe how the user can specify path to `west` using `-DWEST=<path-to-west>`

File: doc/nrf/gs_programming.rst [hunk 5]
```diff
@@ -29,28 +29,29 @@ You can download it from the following links:
...
+#. Select **File -> Open nRF Connect SDK Project**.
 
-    .. figure:: images/ses_open.JPG
+    .. figure:: images/ses_open.png
        :alt: Open nRF Connect SDK Project menu
 
        Open nRF Connect SDK Project menu
 
-#. To import a project into SES, you must specify the following paths:
+#. The first time you import an |NCS| project, SES will prompt you to set the paths to the Zephyr Base directory and the GNU ARM Embedded Toolchain.
```
Comment (tejlmand) [inline doc/nrf/gs_programming.rst]: Should we provide a screen dump here as well ?

File: doc/nrf/gs_programming.rst (hunk 2 above)
Comment (sigvartmh) [reply doc/nrf/gs_programming.rst]: Yeah, that's the safest option as you don't have a PATH variable available in this tab

File: doc/nrf/gs_programming.rst (hunk 1 above)
Comment (sigvartmh) [reply doc/nrf/gs_programming.rst]: I would assume so if it's using `BSD lib`  hence requiring `SPM`

File: doc/nrf/gs_programming.rst (hunk 2 above)
Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: If we don't use the global PATH on macOS, how do we specify where to find mergehex?

**Comment by tejlmand** [comment]
> 
> 
> We're missing information about the Tools > Options configuration.
> Which fields are required there? Which are prefilled?
> I can't get it to work here ... @tejlmand ?

`Tools -> Options -> nRF Connect` contains options relevant for NCS.

The options to set:
- CMake Options
- ARM gcc
- Zephyr base
- CMake
- DTC
- Ninja
- Python
- No of log lines

File: doc/nrf/gs_programming.rst (hunk 1 above)
Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: updated

File: doc/nrf/gs_programming.rst (hunk 2 above)
Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: Added

File: doc/nrf/gs_programming.rst (hunk 3 above)
Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: ok, not changed

File: doc/nrf/gs_programming.rst (hunk 4 above)
Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: added

File: doc/nrf/gs_programming.rst (hunk 5 above)
Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: added

**Comment by ru-fu** [comment]
Thanks, updated.
Can you please check again (most importantly the troubleshooting section)?

File: doc/nrf/gs_programming.rst [hunk 6]
```diff
@@ -98,8 +105,34 @@ it might return an error indicating a project load failure. For example::
...
-error pop-up message and then inspect the text output in SES.
+This issue might be caused by a variety of problems, such as incorrectly specified project file paths.
+SES helps you to identify the source of the issue by providing a text output with detailed information about the error.
+Make sure to click **OK** on the error pop-up message and then inspect the text output in SES.
+
+Missing executables
+===================
+
+On Windows and Linux, SES uses the PATH variable to find executables.
+If you get an error that a tool or command cannot be found, add its location to the PATH variable.
```
Comment (tejlmand) [inline doc/nrf/gs_programming.rst]: How about (feel free to rephrase):
```suggestion
If you get an error that a tool or command cannot be found, first make sure the tool is installed, and if it is already installed then add its location to the PATH variable.
```

**Comment by ru-fu** [comment]
Thanks! Updated as suggested, and I also added a note that the macOS settings are prefilled starting with SES 4.17.

File: doc/nrf/gs_programming.rst [hunk 7]
```diff
@@ -104,8 +111,39 @@ it might return an error indicating a project load failure. For example::
...
+
+On macOS, the global PATH variable is used only if you start SES from the command line.
+If you start SES by running the file :file:`bin/emStudio`, the global PATH is not used, and you must specify the path to all executables under **Tools -> Options** (select the **nRF Connect** tab, see :ref:`ses_options_figure`).
+
+In addition, specify the path to the west tool as additional CMake option, replacing *path_to_west* with the path to the west executable:
+
+.. parsed-literal::
+   :class: highlight
+
+   -D WEST=\ *path_to_west*
```
Comment (carlescufi) [inline doc/nrf/gs_programming.rst]: I don't think you need a space after the `-D`? and why is there a backslash?

I think this should be something like: `-DWEST=path/to/west`

Comment (carlescufi) [reply doc/nrf/gs_programming.rst]: Also, @sigvartmh could we give a reasonable default for the path? Typically if you've installed Python 3 with brew, this should be `/usr/local/bin/west`

Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: The space after the -D is optional - shouldn't hurt to have it.
We need the backslash space to display path_to_west in italics (which shows it's a variable that must be replaced).

Comment (sigvartmh) [reply doc/nrf/gs_programming.rst]: I agree a reasonable default would be  `/usr/local/bin/west` as that would be the default path for most users of `west` when installed through *pip*

Comment (ru-fu) [reply doc/nrf/gs_programming.rst]: added

**Pull request merged**